tools/depends/configure.ac

```diff
@@ -320,45 +320,65 @@ case $host in
         MC_CHECK_NOT_CPU([$use_cpu], "*86")
 
         # setup which sdk to use
+
+        # If tvOS
         if test "$use_platform" = "tvos"; then
+          platform_os="tvos"
```

Comment:
Why make tvOS a complete separate platform instead of a special variant of the iOS platform, like it was done before?

Reply:
Memphiz thought it was a good idea to separate thing concerning iOS and tvOS (ref: https://forum.kodi.tv/showthread.php?tid=312805&pid=2842516#pid2842516)

Reply:
@Memphiz, do you have a preference on setting the OS to tvos or leaving it as ios and using target platform to differentiate instead (similar to your yab branch).

Reply:
Well I have no idea how similar all this is on the cmake side. I would have split cmake side because iOS and tvOS have different platform implementations for entry point and controller and some other classes. I’d say go for the route with lesser if deferi needed. Maybe @Rechi has some more input why separation is not a good idea here.
iOS is a touch device os and tvOS is a set top box os while there are some similarities there also a lot of differences.

Reply:
The reason I'm asking is the following comment from `TargetConditionals.h` in the platforms SDKs. Also most of the build system code looks the same. I was thinking to handle it like the different windowing for linux. 
```
TARGET_OS_* 
These conditionals specify in which Operating System the generated code will
run.  Indention is used to show which conditionals are evolutionary subclasses.  

The MAC/WIN32/UNIX conditionals are mutually exclusive.
The IOS/TV/WATCH conditionals are mutually exclusive.


    TARGET_OS_WIN32           - Generated code will run under 32-bit Windows
    TARGET_OS_UNIX            - Generated code will run under some Unix (not OSX) 
    TARGET_OS_MAC             - Generated code will run under Mac OS X variant
       TARGET_OS_OSX          - Generated code will run under OS X devices
       TARGET_OS_IPHONE          - Generated code for firmware, devices, or simulator
          TARGET_OS_IOS             - Generated code will run under iOS 
          TARGET_OS_TV              - Generated code will run under Apple TV OS
          TARGET_OS_WATCH           - Generated code will run under Apple Watch OS
          TARGET_OS_BRIDGE          - Generated code will run under Bridge devices
       TARGET_OS_SIMULATOR      - Generated code will run under a simulator
       TARGET_OS_EMBEDDED       - Generated code for firmware
   
    TARGET_IPHONE_SIMULATOR   - DEPRECATED: Same as TARGET_OS_SIMULATOR
    TARGET_OS_NANO            - DEPRECATED: Same as TARGET_OS_WATCH

  +------------------------------------------------+
  |                TARGET_OS_MAC                   |
  | +---+  +-------------------------------------+ |
  | |   |  |          TARGET_OS_IPHONE           | |
  | |OSX|  | +-----+ +----+ +-------+ +--------+ | |
  | |   |  | | IOS | | TV | | WATCH | | BRIDGE | | |
  | |   |  | +-----+ +----+ +-------+ +--------+ | |
  | +---+  +-------------------------------------+ |
  +------------------------------------------------+
 ```

Reply:
Thank you this information. I understand better this questioning.

We revert changes to have the most common things between iOS and tvOS.